src/main/java/io/oneko/docker/DockerRegistryPolling.java

```diff
@@ -67,23 +72,41 @@
 	@Scheduled(fixedDelay = 20000, initialDelay = 20000)
 	protected void checkDockerForNewImages() {
 		if (!this.pollingInProgress.getAndSet(true)) {
+			log.trace("Starting polling job");
+			lastPollingJobStartDate = Instant.now();
 			final Mono<List<ProjectMesh>> meshesMono = projectMeshRepository.getAll().
 					collectList();
 			final Mono<List<Project>> projectsMono = projectRepository.getAll()
 					.collectList();
-			meshesMono.zipWith(projectsMono)
-					.subscribe(pair -> this.checkDockerForNewImages(pair.getT1(), pair.getT2()));
+			this.currentPollingJob = meshesMono.zipWith(projectsMono)
+					.flatMap(pair -> this.checkDockerForNewImages(pair.getT1(), pair.getT2()))
+					.subscribe();
+		} else if (shouldPollingJobBeTimedOut() || this.currentPollingJob.isDisposed()) {
+			log.info("Timing out polling job.");
```

Comment:
The job actually yet timed out so we cancel it here.

Reply:
Fixed